python/lsst/pipe/tasks/interpImage.py

```diff
@@ -262,3 +272,172 @@ def interpolateImage(self, maskedImage, psf, defectList, fallbackValue):
+    image: afwImage.MaskedImage,
+    maskPlanes: Iterable[str],
+    *,
+    max_window_extent: lsst.geom.Extent2I,
+    badpix: set | None = None,
+    goodpix: dict | None = None,
+):
+    """Find the location of bad pixels, and neighboring good pixels.
+
+    Parameters
+    ----------
+    image : `~lsst.afw.image.MaskedImage`
+        Image from which to find the bad and the good pixels.
+    maskPlanes : `list` [`str`]
+        List of mask planes to consider as bad pixels.
+    max_window_extent : `lsst.geom.Extent2I`
+        Maximum extent of the window around a bad pixel to consider when
+        looking for good pixels.
+    badpix : `list` [`tuple` [`int`, `int`]], optional
+        A known list of bad pixels. If provided, the function does not look for
+        any additional bad pixels, but it verifies that the provided
+        coordinates correspond to bad pixels. If an input``badpix`` is not
+        found to be bad as specified by ``maskPlanes``, an exception is raised.
+    goodpix : `dict` [`tuple` [`int`, `int`], `float`], optional
+        A known mapping of the coordinates of good pixels to their values, to
+        which any newly found good pixels locations will be added, and the
+        values (even for existing items) will be updated.
+
+    Returns
+    -------
+    badpix : `list` [`tuple` [`int`, `int`]]
+        The coordinates of the bad pixels. If ``badpix`` was provided as an
+        input argument, the returned quantity is the same as the input.
+    goodpix : `dict` [`tuple` [`int`, `int`], `float`]
+        Updated mapping of the coordinates of good pixels to their values.
+
+    Raises
+    ------
+    ValueError
+        If an input ``badpix`` is not found to be bad as specified by
+        ``maskPlanes``.
+    """
+
+    bbox = image.getBBox()
+    if badpix is None:
+        iterator = product(range(bbox.minX, bbox.maxX + 1), range(bbox.minY, bbox.maxY + 1))
+        badpix = set()
+    else:
+        iterator = badpix
+
+    if goodpix is None:
+        goodpix = {}
+
+    for x, y in iterator:
+        if image.mask[x, y] & afwImage.Mask.getPlaneBitMask(maskPlanes):
+            badpix.add((x, y))
+            window = lsst.geom.Box2I.makeCenteredBox(
+                center=lsst.geom.Point2D(x, y),  # center has to be a Point2D instance.
+                size=max_window_extent,
+            )
+            # Restrict to the bounding box of the image.
+            window.clip(bbox)
+
+            for xx, yy in product(range(window.minX, window.maxX + 1), range(window.minY, window.maxY + 1)):
+                if not (image.mask[xx, yy] & afwImage.Mask.getPlaneBitMask(maskPlanes)):
+                    goodpix[(xx, yy)] = image.image[xx, yy]
+        elif (x, y) in badpix:
+            # If (x, y) is in badpix, but did not get flagged as bad,
+            # raise an exception.
+            raise ValueError(f"Pixel ({x}, {y}) is not bad as specified by maskPlanes {maskPlanes}")
```

Comment:
This function ignores entries in `goodpix` (it silently overwrites them with the true image value), but raises if the image doesn't match the `badpix`.  Shouldn't these cases be treated the same way?
While thinking through this case, I realized that `goodpix` is also susceptible to poisoning: an entry passed in `goodpix` will be used even if added to `badpix`.  That definitely needs fixed, even if the rest of the existing logic stays as is (which I'm fine with).

Reply:
The use case here is we have a collection of images that have identical mask planes, but different image planes. So we shouldn't be treating `badpix` (that comes from mask) in the same way as `goodpix` (that comes from both the mask and image planes) the same way. But yes, agree that we should check that `goodpix` and `badpix` are mutually exclusive.